MANIFEST.in

```diff
@@ -1,7 +1,27 @@
-include README.rst
-include MANIFEST.in
-include LICENSE
+# Non Code related contents
+include LICENSE *.rst *.md *.yml *.toml Makefile *.cfg
+graft .github
+
+# Release
+include release.py
 include setup.py
 
-recursive-exclude * __pycache__
-recursive-exclude * *.py[co]
+# Requirements
+include *.txt
+
+# Tests
+include tox.ini .coveragerc
+graft tests
+
+# Examples
+graft examples
+
+# Documentation
+graft docs
+prune docs/_build
+
+# Docker setup
```

Comment:
I'd just include `README`, `LICENSE` and `setup.py`. I don't think we need a lot of those file packaged up and upload.

Reply:
@yunstanford For `whl` file I am trying to keep the bare minimum requirements as you mentioned but for `sdist` it's a bit more. I will let you run tests, and then build things with the `tar` ball.

Reply:
@vltr We should be able to leverage `test_requires` so that when you run `python setup.py test` will set up the test requirements before running the test. 

As for tests, we do something like this. 

```python
from setuptools.command.test import test as TestCommand
class PyTest(TestCommand):
    user_options = [('pytest-args=', 'a', "Arguments to pass to pytest")]

    def initialize_options(self):
        TestCommand.initialize_options(self)
        self.pytest_args = ''

    def run_tests(self):
        import shlex
        import pytest
        errno = pytest.main(shlex.split(self.pytest_args))
        sys.exit(errno)


setup(**kwargs, cmdcls={"test": PyTest})
```

Reply:
@harshanarayana I think that would be great. Thoughts, anyone, to be democratic? :wink:

Reply:
@yunstanford Your suggestions to reduce the manifest contents has been addressed in the new change.

Reply:
> @harshanarayana I think that would be great. Thoughts, anyone, to be democratic? 😉

@huge-success/sanic-core-devs  I added the suggested changes related to `setup.py` based `test` command into the new commit with few helper `make` commands. Take a look at it and let me know if it needs any change.

Reply:
@ahopkins Consider it done. I've added a `make beautify` with optional `sort_imports=1 include_tests=1` to enable fixing import order using `isort`. It also has a simple `make black` and `make isort` task as well.